proxy/fuse/fuse.go

```diff
@@ -332,56 +307,76 @@ func (r *fsRoot) listenerLifecycle(l net.Listener, instance, path string) {
 	}
 }
 
-// ReadDirAll returns a list of files contained in the root directory.
-// It contains a README file which explains how to use the directory.
-// In addition, there will be a file for each instance to which the
-// proxy is actively connected.
-func (r *fsRoot) ReadDirAll(context.Context) ([]fuse.Dirent, error) {
-	ret := []fuse.Dirent{
-		{Name: "README", Type: fuse.DT_File},
+// Readdir implements fs.NodeReaddirer and returns a list of files for each
+// instance to which the proxy is actively connected. In addition, the list
+// includes a README.
+func (r *fsRoot) Readdir(ctx context.Context) (fs.DirStream, syscall.Errno) {
+	activeConns := r.connset.IDs()
+	entries := []fuse.DirEntry{
+		{Name: "README", Mode: 0777 | fuse.S_IFREG},
 	}
-
-	for _, v := range r.connset.IDs() {
-		ret = append(ret, fuse.Dirent{Name: v, Type: fuse.DT_Socket})
+	for _, conn := range activeConns {
+		entries = append(entries, fuse.DirEntry{
+			Name: conn,
+			Mode: 0777 | syscall.S_IFSOCK,
+		})
 	}
-
-	return ret, nil
+	ds := fs.NewListDirStream(entries)
+	return ds, fs.OK
 }
 
-// readme implements the REAME file found in the mounted folder. It is a
-// static read-only text file.
-type readme struct{}
+// readme represents a static read-only text file.
+type readme struct {
+	fs.Inode
+}
```

Comment:
Can you explain why we are using `fs.Inode` here instead of in the interface below? (it doesn't look like any of the functions actually use `readme` so I'm curious we we're using a pointer).